agent/agent.go

```diff
@@ -876,6 +877,15 @@ func (a *agent) run(ctx context.Context) error {
 		return nil
 	})
 
+	eg.Go(func() error {
+		a.logger.Debug(egCtx, "running send logs loop")
+		err := a.logSender.sendLoop(egCtx, aAPI)
```

Comment:
Like I mentioned in my previous review, we need sendLoop running whilst shutting down the agent (`(*agent).Close()`). But I'm pretty sure the context used here will be closed sooner, e.g. when receiving an interrupt, so we will never send logs from shutdown scripts.

Perhaps it would be good to add a test for this case?